### PR TITLE
EVG-5415: Log out all user sessions

### PR DESCRIPTION
**Pull request by ybrill**

### CODE REVIEW

File: model/user/user.go [hunk 1]
```diff
@@ -241,3 +241,25 @@ func GetLoginCache(token string, expireAfter time.Duration) (gimlet.User, bool,
 	}
 	return u, true, nil
 }
+
+// ClearLoginCache clears a user or all users' tokens from the cache, forcibly logging them out
+func ClearLoginCache(user gimlet.User, all bool) error {
+	update := bson.M{"$unset": bson.M{LoginCacheKey: 1}}
+	updatee := bson.M{}
```
Comment (bsamek) [inline model/user/user.go]: The convention is to call this `query`.

File: model/user/user.go [hunk 2]
```diff
@@ -241,3 +241,25 @@ func GetLoginCache(token string, expireAfter time.Duration) (gimlet.User, bool,
...
+		if err != nil {
+			return errors.Wrap(err, "problem finding user by id")
+		}
+		if u == nil {
+			return errors.Errorf("no user '%s' found", user.Username())
+		}
+		updatee = bson.M{IdKey: u.Id}
+	}
+
+	if err := UpdateAll(updatee, update); err != nil {
```
Comment (bsamek) [inline model/user/user.go]: This makes me a bit nervous. To be more conservative, could you instead have an if/else, one branch of which runs `UpdateOne` and the other `UpdateAll`?

File: model/user/user.go [hunk 3]
```diff
@@ -241,3 +241,25 @@ func GetLoginCache(token string, expireAfter time.Duration) (gimlet.User, bool,
...
 }
+
+// ClearLoginCache clears a user or all users' tokens from the cache, forcibly logging them out
+func ClearLoginCache(user gimlet.User, all bool) error {
+	update := bson.M{"$unset": bson.M{LoginCacheKey: 1}}
+	updatee := bson.M{}
+	if !all {
+		u, err := FindOneById(user.Username())
+		if err != nil {
+			return errors.Wrap(err, "problem finding user by id")
```
Comment (bsamek) [inline model/user/user.go]: Include `user.Username()` in the error message. (Use `Wrapf`.)

File: model/user/user_test.go [hunk 4]
```diff
@@ -253,3 +253,32 @@ func (s *UserTestSuite) TestGetLoginCache() {
 	s.False(valid)
 	s.Nil(u)
 }
+
+func (s *UserTestSuite) TestClearLoginCache() {
```
Comment (bsamek) [inline model/user/user_test.go]: This test should insert 2 users, and verify that only 1 is deleted in the `ClearLoginCache(u, false)` case, and that both are deleted in the `ClearLoginCache(u, true)` case.

File: public/static/js/settings.js [hunk 5]
```diff
@@ -66,6 +67,19 @@ mciModule.controller('SettingsCtrl', ['$scope', '$http', '$window', 'notificatio
       });
   }
 
+  $scope.clearToken = function(){
+    if(!confirm("This will log out the current user from all existing sessions. Continue?"))
```
Comment (bsamek) [inline public/static/js/settings.js]: "This will log _you_ out"...

File: model/user/user.go (hunk 1 above)
Comment (ybrill) [reply model/user/user.go]: Done

File: model/user/user.go (hunk 2 above)
Comment (ybrill) [reply model/user/user.go]: Done.

File: model/user/user.go (hunk 3 above)
Comment (ybrill) [reply model/user/user.go]: Done.

File: model/user/user_test.go (hunk 4 above)
Comment (ybrill) [reply model/user/user_test.go]: For ClearLoginCache(nil, true) it now verifies it worked with a representative user.

File: public/static/js/settings.js (hunk 5 above)
Comment (ybrill) [reply public/static/js/settings.js]: Done.

**Pull request merged**